R/utilities-quantity.R

```diff
@@ -162,11 +162,28 @@ setQuantityValuesByPath <- function(quantityPaths, values, simulation, units = N
       # in necessary
       if (!is.null(units[[i]])) {
         mw <- simulation$molWeightFor(path)
-        value <- toBaseUnit(
-          quantityOrDimension = dimension,
-          values = value,
-          unit = units[[i]],
-          molWeight = mw
+        # Try to convert to base unit. If the provided unit is not valid for the
+        # quantity dimension, throw an error
+        tryCatch(
+          {
+            value <- toBaseUnit(
+              quantityOrDimension = dimension,
+              values = value,
+              unit = units[[i]],
+              molWeight = mw
+            )
+          },
+          error = function(e) {
+            stop(paste0(
+              messages$wrongUnitForQuantity(
+                quantityPath = path,
+                unit = units[[i]]
+              ),
+              "\n",
```

Comment:
is this message useful? it's cleary a conversion error no?

Reply:
I was setting parameter values from excel file using `setParameterValuesByPath()` and it was failing because I forgot to define the unit. With the previous message

> Message: Unit '' is not defined in dimension 'Volume'.

I had to debug to find what exactly was wrong. If I've had the path of the quantity (what is now in the improved message), I would have have much better idea where this is coming from.